src/modal/modalcomponents/ModalId.tsx

```diff
@@ -0,0 +1,40 @@
+import share from "../../image/share.png";
+import rename from "../../image/namechange.png";
+import Delete from "../../image/delete.png";
+
+type ModalID = {
+  addFolder: string;
+  share: string;
+  rename: string;
+  delete: string;
+};
```

Comment:
P3: 코드가 반복되니 아래처럼도 가능합니다.

```suggestion
type ModalIDTypes = [ 'addFolder', 'share', 'rename', 'delete' ]
type ModalID ={ [k in ModalIDTypes]: string } 
```